runners/core-java/src/main/java/org/apache/beam/runners/core/triggers/TriggerStateMachines.java

```diff
@@ -171,18 +174,38 @@ private TriggerStateMachine evaluateSpecific(OrFinallyTrigger v) {
     }
 
     private OnceTriggerStateMachine evaluateSpecific(AfterProcessingTime v) {
-      return evaluateSpecific((AfterDelayFromFirstElement) v);
-    }
-
-    private OnceTriggerStateMachine evaluateSpecific(final AfterDelayFromFirstElement v) {
       return new AfterDelayFromFirstElementStateMachineAdapter(v);
     }
 
     private static class AfterDelayFromFirstElementStateMachineAdapter
         extends AfterDelayFromFirstElementStateMachine {
 
-      public AfterDelayFromFirstElementStateMachineAdapter(AfterDelayFromFirstElement v) {
-        this(v.getTimeDomain(), v.getTimestampMappers());
+      private static final Function<TimestampTransform, SerializableFunction<Instant, Instant>>
+          CONVERT_TIMESTAMP_TRANSFORM =
+              new Function<TimestampTransform, SerializableFunction<Instant, Instant>>() {
+                @Override
+                @Nonnull
```

Comment:
Isn't it for use when overriding a `@Nullable` thing? In the case of the parameter, it indicates that it must not be null, despite the signature (actually an unsafe alteration) while in the case of the return value it indicates an additional guarantee that it won't return null.

TBH this way of hacking nullability onto the type system without real integration just doesn't quite work. I'll happily remove this one.